privacyidea/lib/crypto.py

```diff
@@ -371,47 +326,52 @@ def decryptPassword(cryptPass, convert_unicode=False):
     # takes unicode strings!). But always returning unicode might break
     # other call sites of ``decryptPassword``. So we add the
     # keyword argument to avoid breaking compatibility.
-    from privacyidea.lib.utils import to_unicode
     hsm = get_hsm()
     try:
         ret = hsm.decrypt_password(cryptPass)
-    except Exception as exx:  # pragma: no cover
+    except Exception as exx:
         log.warning(exx)
         ret = FAILED_TO_DECRYPT_PASSWORD
     try:
         if convert_unicode:
-            ret = to_unicode(ret)
-    except Exception as exx:  # pragma: no cover
+            ret = _to_unicode(ret)
+    except Exception as exx:
         log.warning(exx)
         # just keep ``ret`` as a bytestring in that case
     return ret
 
 
 @log_with(log, log_exit=False)
 def decryptPin(cryptPin):
+    """
+
+    :param cryptPin: the encrypted pin
+    :type cryptPin: str, bytes, unicode
+    :return: the decrypted pin
+    :rtype: str
+    """
     hsm = get_hsm()
-    ret = hsm.decrypt_pin(cryptPin)
-    return ret
+    return _to_unicode(hsm.decrypt_pin(_to_bytes(cryptPin)))
 
 
 @log_with(log, log_entry=False)
 def encrypt(data, iv, id=0):
     '''
-    encrypt a variable from the given input with an initialiation vector
-
-    :param input: buffer, which contains the value
-    :type  input: buffer of bytes
-    :param iv:    initilaitation vector
-    :type  iv:    buffer (20 bytes random)
-    :param id:    contains the id of which key of the keyset should be used
-    :type  id:    int
-    :return:      encryted buffer
+    encrypt a variable from the given input with an initialisation vector
 
+    :param data: buffer, which contains the value
+    :type  data: bytes or str
+    :param iv:   initialisation vector
+    :type  iv:   bytes or str
+    :param id:   contains the key id of the keyset which should be used
+    :type  id:   int
+    :return:     encrypted and hexlified data
```

Comment:
Hm, I see why you changed ``encrypt`` to return a hexstring, and I think that makes sense. But I think it's unfortunate that ``encrypt`` and ``decrypt`` are not inverses anymore (because ``decrypt`` expects bytes).
Maybe we can keep the old ``encrypt``/``decrypt`` operating on bytes, and add a new ``encrypt_and_hexlify``? But maybe there's a better way to fix it.

Reply:
Yes, i don't like this either. I'll look into the usage of these functions again, maybe there is a better approach.

Reply:
@fredreichbier : @plettich added a docstring, so that we can handle this later with issue #337

Reply:
We can fix this in a later PR.

Reply:
:+1: